sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnTester.java

```diff
@@ -285,7 +288,13 @@ public void processTimestampedElement(TimestampedValue<InputT> element) throws E
       startBundle();
     }
     try {
-      fn.processElement(createProcessContext(element));
+      final TestProcessContext processContext = createProcessContext(element);
+      fnInvoker.invokeProcessElement(new DoFnInvoker.FakeArgumentProvider<InputT, OutputT>() {
```

Comment:
This is a bit of an abuse of the "is a" relationship with `FakeArgumentProvider`. It is actually not harmless - that argument provider returns `null` instead of throwing when an unsupported method is called.

Probably `FakeArgumentProvider` should be removed or modified to always `throw`. Until then, I'd just make an `ArgumentProvider` especially for the purpose here.